src/submission/models.py

```diff
@@ -364,6 +370,85 @@ def validate(self, value, model_instance):
                     raise
 
 
+class ArticleSearchManager(BaseSearchManagerMixin):
+
+    def search(self, *args, **kwargs):
+        queryset = super().search(*args, **kwargs)
+        return queryset.filter(
+            date_published__lte=timezone.now()
+        )
+
+    def mysql_search(self, search_term, search_filters, site=None):
+        queryset = self.get_queryset().none()
+        if not search_term or not any(search_filters.values()):
+            return queryset
+        querysets = []
+        if search_filters.get('title'):
+            querysets.append(self.get_queryset().filter(title__search=search_term))
+        if search_filters.get('authors'):
+            querysets.append(self.get_queryset().filter(
+                frozenauthor__first_name__search=search_term))
+            querysets.append(self.get_queryset().filter(
+                frozenauthor__last_name__search=search_term))
+        if search_filters.get("abstract"):
+            querysets.append(self.get_queryset().filter(abstract__search=search_term))
+        if search_filters.get('keywords'):
+            querysets.append(self.get_queryset().filter(
+                keywords__word__search=search_term))
+        if search_filters.get("full_text"):
+            querysets.append(self.get_queryset().filter(
+                galley__file__text__contents__search=search_term))
+        for search_queryset in querysets:
+            queryset |= search_queryset
+        return queryset
+
+    def postgres_search(self, search_term, search_filters, site=None):
+        queryset = self.get_queryset()
+        if not search_term or not any(search_filters.values()):
+            return queryset.none()
+        if site:
+            queryset = queryset.filter(journal=site)
+        lookups, annotations = self.build_postgres_lookups(
+            search_term, search_filters)
+        if annotations:
+            queryset = queryset.annotate(**annotations)
+        if lookups:
+            queryset = queryset.filter(**lookups)
+
+        return queryset.order_by("-relevance")
+
+    def build_postgres_lookups(self, search_term, search_filters):
+        lookups = {}
+        annotations = {"relevance": models.Value(1.0, models.FloatField())}
+        vectors = []
+        if search_filters.get('title'):
+            vectors.append(SearchVector('title', weight="A"))
+        if search_filters.get("abstract"):
+            vectors.append(SearchVector('abstract', weight="B"))
+        if search_filters.get('authors'):
+            vectors.append(SearchVector('frozenauthor__last_name', weight="B"))
+            vectors.append(SearchVector('frozenauthor__first_name', weight="B"))
+        if search_filters.get('keywords'):
+            vectors.append(SearchVector('keywords__word', weight="C"))
```

Comment:
This is my personal opinion but we might want to gauge from others on the weighting of Keywords here.